target_chains/ethereum/contracts/contracts/pyth/PythGovernance.sol

```diff
@@ -190,4 +201,44 @@ abstract contract PythGovernance is
 
         emit ValidPeriodSet(oldValidPeriod, validTimePeriodSeconds());
     }
+
+    function setWormholeAddress(
+        SetWormholeAddressPayload memory payload,
+        bytes memory encodedVM
+    ) internal {
+        address oldWormholeAddress = address(wormhole());
+        setWormhole(payload.newWormholeAddress);
+
+        // We want to verify that the new wormhole address is valid, so we make sure that it can
+        // parse and verify the same governance VAA that is used to set it.
+        (IWormhole.VM memory vm, bool valid, ) = wormhole().parseAndVerifyVM(
+            encodedVM
+        );
+
+        if (!valid) revert PythErrors.InvalidGovernanceMessage();
+
+        if (!isValidGovernanceDataSource(vm.emitterChainId, vm.emitterAddress))
```

Comment:
we already do some of these checks in `verifyGovernanceVM` in `executeGovernanceInstruction`

Reply:
This is called via the new wormhole, it's just an additional sanity check that makes sure the new wormhole is parsing messages corerctly.

Reply:
also suggest that the reverts here should return a more specific error, like "InvalidWormholeContract".